### PR TITLE
Added failed test

### PR DESCRIPTION
**Pull request by CvekCoding**
I have added a test which shows my issue - if you (or DI) instantiate basic symfony serializer, it invokes `setSerializer` method of all `SerializerAware` services and rewrites `serializer` property from `DunglasSerializer` to `SymfobySerializer`.
This breaks the further normalization.

### CODE REVIEW

**Comment by CvekCoding** [comment]
The added test is totally the same like first one with only one difference - in a first line I instantiate `SymfonySerializer`. This way I imitated what happens in my project - when Symfony starts, it creates all the listeners, and some of them have `SerializerInterface` dependency.

**Comment by Toflar** [comment]
@ajgarlag is that releated to your changes maybe? 😊

**Comment by ajgarlag** [comment]
Surely! I think this is related to https://github.com/symfony/symfony/issues/28544. @CvekCoding Can you try to revert https://github.com/dunglas/doctrine-json-odm/pull/71/commits/a8c3c0aa97e3a6de80c7f65cc4f323b174b89ff9 and tell us if it works?

**Comment by CvekCoding** [comment]
@ajgarlag yes, this helped, thanks. I'll continue my investigations.

File: src/Bundle/Resources/config/services.xml [hunk 1]
```diff
@@ -5,10 +5,20 @@
            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
 
     <services>
+        <service id="dunglas_doctrine_json_odm.normalizer.raw_object" class="Symfony\Component\Serializer\Normalizer\ObjectNormalizer" public="false">
```
Comment (ajgarlag) [inline src/Bundle/Resources/config/services.xml]:
```suggestion
        <service id="dunglas_doctrine_json_odm.normalizer.object" class="Symfony\Component\Serializer\Normalizer\ObjectNormalizer" public="false">
```

File: src/Bundle/Resources/config/services.xml [hunk 2]
```diff
@@ -5,10 +5,20 @@
...
+        </service>
+
+        <service id="dunglas_doctrine_json_odm.normalizer.array" class="Symfony\Component\Serializer\Normalizer\ArrayDenormalizer" public="false" />
+
         <service id="dunglas_doctrine_json_odm.serializer" class="Dunglas\DoctrineJsonOdm\Serializer" public="true">
             <argument type="collection">
-                <argument type="service" id="serializer.denormalizer.array" />
-                <argument type="service" id="serializer.normalizer.object" />
+                <argument type="service" id="dunglas_doctrine_json_odm.normalizer.array" />
+                <argument type="service" id="dunglas_doctrine_json_odm.normalizer.raw_object" />
```
Comment (ajgarlag) [inline src/Bundle/Resources/config/services.xml]:
```suggestion
                <argument type="service" id="dunglas_doctrine_json_odm.normalizer.object" />
```

**Comment by Toflar** [comment]
Can I ask you to fix the CS issues? Thank you!

File: src/Bundle/Resources/config/services.xml (hunk 2 above)
Comment (CvekCoding) [reply src/Bundle/Resources/config/services.xml]: Done

File: src/Bundle/Resources/config/services.xml (hunk 1 above)
Comment (CvekCoding) [reply src/Bundle/Resources/config/services.xml]: Done

**Comment by CvekCoding** [comment]
> Can I ask you to fix the CS issues? Thank you!

Done

**Comment by CvekCoding** [comment]
Added a few important notes here https://github.com/dunglas/doctrine-json-odm/pull/74/files
Please, look into them.

**Comment by otakupahp** [comment]
Since there is only a conflict in the README.md, could anyone fix this and merge this pull?

**Comment by ajgarlag** [comment]
@Toflar I think this this PR should be merged and tagged as rc2. The README changes should be applied too.

**Comment by Toflar** [comment]
Thank you @CvekCoding and @ajgarlag! I've also tagged rc2! Sorry folks that it took so long!
I set myself a reminder in about 2 weeks time and I'd like to ask you to test everything again and if everything is fine, I'll tag 1.0.0 stable in 2 weeks!

**Comment by Toflar** [comment]
As promised: [Version 1.0.0 stable is here](https://github.com/dunglas/doctrine-json-odm/releases/tag/1.0.0)!

Thanks to everyone involved! You may now update your Composer requirements and require the stable version 😎

**Pull request merged**